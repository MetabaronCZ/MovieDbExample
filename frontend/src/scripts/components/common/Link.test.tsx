import { describe, expect, it } from '@jest/globals';

import { Link } from './Link';
import { getByText } from '@testing-library/dom';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/common/Link', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Link to="/">CONTENT</Link>, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const link = getByText(container, 'CONTENT');
    expect(link).toBeInTheDocument();
    expect(link.nodeName.toLowerCase()).toEqual('a');
  });

  it('should render props', async () => {
    const { container } = await waitForComponent(
      <Link to="/" title="TITLE" external>
        CONTENT
      </Link>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const link = getByText(container, 'CONTENT');
    expect(link).toBeInTheDocument();

    expect(link).toHaveAttribute('href', '/');
    expect(link).toHaveAttribute('title', 'TITLE');
    expect(link).toHaveAttribute('target', '_blank');
  });
});
