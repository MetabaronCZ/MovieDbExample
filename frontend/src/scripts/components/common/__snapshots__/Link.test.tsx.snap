// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Link should render properly 1`] = `
.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
}

.c0:hover {
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

<div>
  <a
    class="c0"
    data-discover="true"
    href="/"
  >
    CONTENT
  </a>
</div>
`;

exports[`components/common/Link should render props 1`] = `
.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
}

.c0:hover {
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

<div>
  <a
    class="c0"
    href="/"
    target="_blank"
    title="TITLE"
  >
    CONTENT
  </a>
</div>
`;
